relay-server/src/utils/rate_limits.rs

```diff
@@ -145,62 +202,41 @@ where
 
     /// Process rate limits for the envelope, removing offending items and returning applied limits.
     pub fn enforce(mut self, envelope: &mut Envelope, scoping: &Scoping) -> Result<RateLimits, E> {
-        self.aggregate(envelope);
-        let rate_limits = self.execute(scoping)?;
-        envelope.retain_items(|item| self.retain_item(item));
-        Ok(rate_limits)
-    }
-
-    fn aggregate(&mut self, envelope: &Envelope) {
-        for item in envelope.items() {
-            if item.creates_event() {
-                self.infer_category(item);
-            } else if item.ty() == ItemType::Attachment {
-                // Plain attachments do not create events.
-                self.has_plain_attachments = true;
-            }
-
-            match item.ty() {
-                ItemType::Attachment => self.attachment_quantity += item.len().max(1),
-                ItemType::Session => self.session_quantity += 1,
-                _ => (),
-            }
+        let mut summary = EnvelopeSummary::compute(envelope);
+        if let Some(event_category) = self.event_category {
+            summary.event_category = Some(event_category);
```

Comment:
Note that this changed. Rather than setting the event category first, and then aggregating, we now aggregate first and then overwrite the `event_category`. Given how `infer_category` is written, this is equivalent.